code/modules/clothing/clothing.dm

```diff
@@ -23,6 +24,29 @@
 	. = ..()
 	gunshot_residue = null
 
+
+//Delayed equipping
+/obj/item/clothing/pre_equip(var/mob/user, var/slot)
+	if (equip_delay > 0)
+		//If its currently worn, we must be taking it off
+		if (is_worn())
+			user.visible_message(
+				SPAN_NOTICE("[user] starts taking off \the [src]..."),
+				SPAN_NOTICE("You start taking off \the [src]...")
+			)
+			if(!do_after(user,equip_delay,src))
+				return TRUE //A nonzero return value will cause the equipping operation to fail
+
+		else if (is_held() && !(slot in unworn_slots))
```

Comment:
`is_held()` is not `is_equip()` and should already check `!(slot in unworn_slots)`

Reply:
I think i don't understand this comment, let me explain what i was doing here though

This section is for checking if the user is trying to put the item on. Equipping an item is defined as it moving from your hands, into a slot which is considered "worn"
Is held specifically checks for it being in the hands which is the first part of that, and checking the list in the second part tells us if the user is attempting to put it into a worn slot